cli/flox-rust-sdk/src/providers/catalog.rs

```diff
@@ -517,8 +550,31 @@ impl ClientTrait for CatalogClient {
         package_groups: Vec<PackageGroup>,
     ) -> Result<Vec<ResolvedPackageGroup>, ResolveError> {
         tracing::debug!(n_groups = package_groups.len(), "resolving package groups");
+
+        let semver_normalized_package_groups = package_groups
+            .into_iter()
+            .map(|group| {
+                let descriptors = group
+                    .descriptors
+                    .into_iter()
+                    .map(|descriptor| PackageDescriptor {
+                        version: descriptor.version.map(|version| {
+                            SemverNormalized::try_from(version.clone())
+                                .map_or(version, |semver| semver.to_string())
```

Comment:
The integration test failures have highlighted that this has a bad side-effect of adding a carat to versions that didn't originally have one (e.g. `1.2.3` -> `^1.2.3`) which changes the mean from `1.2.3.x` to `1.2.x` 🤔